packages/common/src/components/CopyableToClipboard/index.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import Clipboard from 'clipboard';
+import { invokeGlobalErrorHandler } from '../../utilities/splash.screen';
+
+interface IProps {
+  textGetter: () => string;
+  globallyUniqueSelector?: string;
+  onSuccess?: () => void;
+  onError?: () => void;
+}
+
+interface IState {}
+
+class CopyableToClipboard extends React.Component<IProps, IState> {
+  private idIfAny: string;
+  private clipboard: Clipboard;
+
+  constructor(props: IProps) {
+    super(props);
+
+    if (!this.props.globallyUniqueSelector) {
+      this.idIfAny = 'export-to-clipboard-' + CopyableToClipboard.nextGlobalId++;
+    }
+  }
+
+  componentDidMount() {
+    const selector = this.props.globallyUniqueSelector || `#${this.idIfAny} > *`;
+    this.clipboard = new Clipboard(selector, {
+      text: this.props.textGetter,
+    });
+
+    if (this.props.onSuccess) {
```

Comment:
you could probably make good use of defaultProps here for onSuccess and onFailure.

Reply:
What's defaultProps?

Reply:
https://reactjs.org/docs/react-component.html#defaultprops

you can check out editor's Main, the common PivotBar and Console, those are the only ones I can remember off the top of my head that we have. searching `defaultProps` should find the rest.